key-share/src/trusted_dealer.rs

```diff
@@ -97,18 +97,62 @@ impl<E: Curve> TrustedDealerBuilder<E> {
     ///
     /// Returns error if provided inputs are invalid, or if internal
     /// error has occurred.
+    ///
+    /// For Shamir secret sharing, it's shared at points `1` to `n`
+    ///
+    /// Returns error if provided inputs are invalid, or if internal
+    /// error has occurred.
     pub fn generate_shares(
         self,
         rng: &mut (impl rand_core::RngCore + rand_core::CryptoRng),
     ) -> Result<Vec<CoreKeyShare<E>>, TrustedDealerError> {
-        let shared_secret_key = self
-            .shared_secret_key
-            .unwrap_or_else(|| NonZero::<SecretScalar<_>>::random(rng));
-        let shared_public_key = Point::generator() * &shared_secret_key;
         let key_shares_indexes = (1..=self.n)
             .map(|i| generic_ec::NonZero::from_scalar(Scalar::from(i)))
             .collect::<Option<Vec<_>>>()
             .ok_or(Reason::DeriveKeyShareIndex)?;
+        self.generate_shares_at(key_shares_indexes, rng)
+    }
+
+    /// Generates [`CoreKeyShare`]s shared at random points
+    ///
+    /// Returns error if provided inputs are invalid, or if internal
+    /// error has occurred.
+    ///
+    /// For Shamir secret sharing, the points at which the value is shared at
+    /// are chosen at random between `1` and `u16::MAX`. For additive shares,
+    /// this is the same as [`TrustedDealerBuilder::generate_shares`]
+    ///
+    /// Returns error if provided inputs are invalid, or if internal
+    /// error has occurred.
+    pub fn generate_shares_at_random(
+        self,
+        rng: &mut (impl rand_core::RngCore + rand_core::CryptoRng),
+    ) -> Result<Vec<CoreKeyShare<E>>, TrustedDealerError> {
+        let key_shares_indexes =
+            rand::seq::index::sample(rng, usize::from(u16::MAX - 1), usize::from(self.n))
+                .iter()
+                .map(|i| generic_ec::NonZero::from_scalar(Scalar::from(i + 1)))
+                .collect::<Option<Vec<_>>>()
+                .ok_or(Reason::DeriveKeyShareIndex)?;
```

Comment:
I'm super confused. Why not just generate `n` random scalars? Why do you want that all scalars are < u16::MAX ?

Reply:
Hmm, it seems you're right, we don't need them to be related to u16 in any way. A brain fart on my part